alica_engine/include/engine/AlicaTimer.h

```diff
@@ -0,0 +1,57 @@
+#pragma once
+
+#include "engine/AlicaClock.h"
+#include "engine/IAlicaTimer.h"
+
+#include <atomic>
+#include <condition_variable>
+#include <functional>
+#include <memory>
+#include <stdexcept>
+#include <thread>
+
+namespace alica
+{
+
+class AlicaSystemTimer : public alica::IAlicaTimer
+{
+public:
+    using TimerCb = std::function<void()>;
+
+    AlicaSystemTimer(TimerCb&& userCb, alica::AlicaTime period);
+    ~AlicaSystemTimer();
+
+private:
+    class AlicaSystemTimerImpl;
+    std::shared_ptr<AlicaSystemTimerImpl> _impl;
```

Comment:
can't this be a unique_ptr?

Reply:
Is it okay for timer to be non copyable?

Reply:
Having a copy of timer sounds suspicious a little bit

Reply:
yes, I think the engine does not assume that Timer's are copyable. I also agree with ntfshard that a copyable timer is less desirable (although ros does not seem to think so)

Reply:
Yeah non copyable is fine at long as the tests pass with it

Reply:
yeah, impl as unique_ptr in pimpl idiom seems more common, but I left this one from the old guy because I didn't want to change much besides the goal of the task which was getting rid of ros. I can change.